### PR TITLE
Delete compose when timeout

### PR DESCRIPTION
**Pull request by tkdchen**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

**Review by MartinBasti — APPROVED** [review]

File: atomic_reactor/plugins/pre_resolve_composes.py [hunk 1]
```diff
@@ -91,21 +92,30 @@ def __init__(self, tasker, workflow,
...
+            return
+
+        self.adjust_compose_config()
+        self.request_compose_if_needed()
+        try:
+            self.wait_for_composes()
+        except WaitComposeToFinishTimeout as e:
+            self.log.info(str(e))
+            self.log.info('Canceling the compose %s', e.compose_id)
+            self.odcs_client.cancel_compose(e.compose_id)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_resolve_composes.py:114]: We should cancel all of the not-yet-completed composes.

Comment (lkolacek) [reply atomic_reactor/plugins/pre_resolve_composes.py:114]: +1

**Review by ben-alkov — APPROVED** [review]

File: atomic_reactor/plugins/pre_resolve_composes.py (hunk 1 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_resolve_composes.py:114]: I'd go with merging this one and extending it in another PR

Comment (lkolacek) [reply atomic_reactor/plugins/pre_resolve_composes.py:114]: Sounds reasonable.

**Review by lkolacek — APPROVED** [review]

**Comment by MartinBasti** [comment]
Merged, release notes will be part of the followup PR

**Pull request merged**